tests/PySys/software-management-end-to-end/environment_sm_management.py

```diff
@@ -0,0 +1,217 @@
+
+        if self.myPlatform != "container":
+            self.skipTest("Testing the apt plugin is not supported on this platform")
+
+        tenant = self.project.tenant
+        user = self.project.username
+        password = self.project.c8ypass
+
+        # Place to save the id of the operation that we started.
+        # This is suitable for one operation and not for multiple ones running
+        # at the same time.
+        self.operation_id = None
+
+        auth = bytes(f"{tenant}/{user}:{password}", "utf-8")
+        self.header = {
+            b"Authorization": b"Basic " + base64.b64encode(auth),
+            b"content-type": b"application/json",
+            b"Accept": b"application/json",
+        }
+
+        # Make sure we have no last operations pending
+        self.wait_until_succcess()
+
+    def trigger_action(self, package_name, package_id, version, url, action):
+        """Trigger a installation or deinstallation of a package"""
+
+        self.trigger_action_json(
+            [
+                {
+                    "id": package_id,
+                    "name": package_name,
+                    "version": version,
+                    "url": url,
+                    "action": action,
+                }
+            ]
+        )
+
+    def trigger_action_json(self, json_content):
+        """Take an actions description that is then forwarded to c8y"""
+
+        url = "https://thin-edge-io.eu-latest.cumulocity.com/devicecontrol/operations"
+
+        payload = {
+            "deviceId": self.project.deviceid,
+            "description": "Apply software changes, triggered from PySys test",
+            "c8y_SoftwareUpdate": json_content,
+        }
+
+        req = requests.post(url, json=payload, headers=self.header)
+
+        jresponse = json.loads(req.text)
+
+        self.log.info(f"Response status: {req.status_code}")
+        self.log.info(f"Response to action: {json.dumps(jresponse, indent=4)}")
+        self.log.info(f"Started operation: {jresponse['id']}")
+
+        self.operation = jresponse
+        self.operation_id = jresponse["id"]
+
+        if req.status_code != 201:  # Request was accepted
+            raise SystemError("Got HTTP status %s", req.status_code)
+
+    def is_status_fail(self):
+        if self.operation_id:
+            return self.check_status_of_operation("FAILED")
+        else:
+            return self.check_last_status("FAILED")
+
+    def is_status_success(self):
+        if self.operation_id:
+            return self.check_status_of_operation("SUCCESSFUL")
+        else:
+            return self.check_last_status("SUCCESSFUL")
+
+    def check_last_status(self, status):
+        """Check if the last operation is successfull
+        """
+
+        params = {
+            "deviceId": self.project.deviceid,
+            "pageSize": 10,
+            "revert": "true",
+        }
+
+        url = "https://thin-edge-io.eu-latest.cumulocity.com/devicecontrol/operations"
```

Comment:
Would be nice if this would be read from env variable such that we can run it from local machines with custom tenants.